ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -519,15 +540,19 @@ private void processReplicaMetadataResponse(Set<StoreKey> missingStoreKeys,
             "Blob id is not in the expected partition Actual partition " + blobId.getPartition()
                 + " Expected partition " + remoteReplicaInfo.getLocalReplicaId().getPartitionId());
       }
-      if (!missingStoreKeys.contains(messageInfo.getStoreKey())) {
+      if (!remoteToLocal.containsKey(messageInfo.getStoreKey())) {
+        throw new IllegalStateException("remoteToLocal does not contain key "+messageInfo.getStoreKey());
+      }
+      BlobId localKey = (BlobId) remoteToLocal.get(messageInfo.getStoreKey());
+      if (!missingStoreKeys.contains(messageInfo.getStoreKey()) && localKey != null) {
```

Comment:
get rid of the second condition?

Reply:
can still run into null localKeys because it is scanning through the keys that are in the replica.  unit test fails when 2nd condition is removed